mlflow/models/model.py

```diff
@@ -698,8 +698,17 @@ def log(
                 model_config = kwargs["model_config"]
                 if isinstance(model_config, str):
                     try:
-                        with open(model_config) as f:
-                            model_config = json.load(f)
+                        file_extension = os.path.splitext(model_config)[1].lower()
```

Comment:
Adding more logic to check types and load things correctly based on this PR: https://github.com/mlflow/mlflow/pull/11880